src/twisted/python/threadpool.py

```diff
@@ -9,15 +9,33 @@
 instead of creating a thread pool directly.
 """
 
+from __future__ import annotations
+
 from threading import Thread, current_thread
-from typing import List
+from typing import Any, Callable, List, Optional, TypeVar
+
+from typing_extensions import ParamSpec, Protocol, TypedDict
 
 from twisted._threads import pool as _pool
 from twisted.python import context, log
 from twisted.python.deprecate import deprecated
 from twisted.python.failure import Failure
 from twisted.python.versions import Version
 
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+
+
+class _SupportsQsize(Protocol):
+    def qsize(self) -> int:
```

Comment:
Can we use the zope.Interface here ?

I don't have experience with Python typing...so maybe this is ok.

```suggestion
class _SupportsQsize(Protocol):
    """
    This is only used to define the interface.
    """
    def qsize(self) -> int:  # pragma: no cover
```

Reply:
I think there's a soft preference for protocols unless Zope.Interface's features (adaptation, explicit opt-in) are helpful in a certain situation. See https://blog.glyph.im/2021/03/interfaces-and-protocols.html for lots of details